### ISSUE TITLE
print-profile-events not always do proper summary

### ISSUE THREAD

**Issue by filimonov** [ClickHouse/ClickHouse#34719]
With ` --profile-events-delay-ms=-1` i expect to see only `[ 0 ]` (i.e. summary) rows in the output.

But for some queries, it prints also some per-thread stats.

```
clickhouse-client --query='CREATE TABLE test.aaaa Engine = MergeTree ORDER BY (ac, nw) AS SELECT toUInt64(rand(1) % 20000000) as ac, toFloat32(1) as wg, toUInt16(rand(3) % 400) as nw FROM numbers_mt(10000000)'

clickhouse-client --print-profile-events  --profile-events-delay-ms=-1  --query='SELECT  nw, sum(WR) AS R FROM ( SELECT AVG(wg) AS WR, ac, nw FROM test.aaaa GROUP BY ac, nw ) GROUP BY nw ORDER BY R DESC LIMIT 10'
323	25480
391	25417
314	25391
295	25390
76	25362
68	25326
203	25323
34	25318
98	25272
330	25269
[laptop-5591] 2022.02.18 11:26:11 [ 3022 ] ReadBufferFromFileDescriptorRead: 25 (increment)
[laptop-5591] 2022.02.18 11:26:11 [ 3022 ] ReadBufferFromFileDescriptorReadBytes: 9774080 (increment)
[laptop-5591] 2022.02.18 11:26:11 [ 3022 ] ReadCompressedBytes: 9683553 (increment)
[laptop-5591] 2022.02.18 11:26:11 [ 3022 ] CompressedReadBufferBlocks: 371 (increment)
[laptop-5591] 2022.02.18 11:26:11 [ 3022 ] CompressedReadBufferBytes: 24313856 (increment)
[laptop-5591] 2022.02.18 11:26:11 [ 3022 ] OpenedFileCacheHits: 3 (increment)
[laptop-5591] 2022.02.18 11:26:11 [ 3022 ] IOBufferAllocs: 6 (increment)
[laptop-5591] 2022.02.18 11:26:11 [ 3022 ] IOBufferAllocBytes: 1283047 (increment)
[laptop-5591] 2022.02.18 11:26:11 [ 3022 ] ArenaAllocChunks: 12 (increment)
[laptop-5591] 2022.02.18 11:26:11 [ 3022 ] ArenaAllocBytes: 33546240 (increment)
[laptop-5591] 2022.02.18 11:26:11 [ 3022 ] MarkCacheHits: 3 (increment)
[laptop-5591] 2022.02.18 11:26:11 [ 3022 ] CreatedReadBufferOrdinary: 3 (increment)
[laptop-5591] 2022.02.18 11:26:11 [ 3022 ] SelectedRows: 1736208 (increment)
[laptop-5591] 2022.02.18 11:26:11 [ 3022 ] SelectedBytes: 24306912 (increment)
[laptop-5591] 2022.02.18 11:26:11 [ 3022 ] ContextLock: 1 (increment)
[laptop-5591] 2022.02.18 11:26:11 [ 3022 ] RealTimeMicroseconds: 239654 (increment)
[laptop-5591] 2022.02.18 11:26:11 [ 3022 ] UserTimeMicroseconds: 190158 (increment)
[laptop-5591] 2022.02.18 11:26:11 [ 3022 ] SystemTimeMicroseconds: 47825 (increment)
[laptop-5591] 2022.02.18 11:26:11 [ 3022 ] SoftPageFaults: 33870 (increment)
[laptop-5591] 2022.02.18 11:26:11 [ 3022 ] OSCPUVirtualTimeMicroseconds: 237976 (increment)
[laptop-5591] 2022.02.18 11:26:11 [ 3022 ] OSReadChars: 9081856 (increment)
[laptop-5591] 2022.02.18 11:26:11 [ 3022 ] OSWriteChars: 10240 (increment)
[laptop-5591] 2022.02.18 11:26:11 [ 3022 ] QueryProfilerRuns: 1 (increment)
[laptop-5591] 2022.02.18 11:26:11 [ 3022 ] MemoryTrackerUsage: 73276641 (gauge)
[laptop-5591] 2022.02.18 11:26:11 [ 11393 ] Seek: 6 (increment)
[laptop-5591] 2022.02.18 11:26:11 [ 11393 ] ReadBufferFromFileDescriptorRead: 35 (increment)
[laptop-5591] 2022.02.18 11:26:11 [ 11393 ] ReadBufferFromFileDescriptorReadBytes: 12901995 (increment)
[laptop-5591] 2022.02.18 11:26:11 [ 11393 ] ReadCompressedBytes: 10197959 (increment)
[laptop-5591] 2022.02.18 11:26:11 [ 11393 ] CompressedReadBufferBlocks: 378 (increment)
[laptop-5591] 2022.02.18 11:26:11 [ 11393 ] CompressedReadBufferBytes: 24772174 (increment)
[laptop-5591] 2022.02.18 11:26:11 [ 11393 ] OpenedFileCacheHits: 3 (increment)
[laptop-5591] 2022.02.18 11:26:11 [ 11393 ] IOBufferAllocs: 12 (increment)
[laptop-5591] 2022.02.18 11:26:11 [ 11393 ] IOBufferAllocBytes: 2852762 (increment)
[laptop-5591] 2022.02.18 11:26:11 [ 11393 ] ArenaAllocChunks: 12 (increment)
[laptop-5591] 2022.02.18 11:26:11 [ 11393 ] ArenaAllocBytes: 33546240 (increment)
[laptop-5591] 2022.02.18 11:26:11 [ 11393 ] MarkCacheHits: 6 (increment)
[laptop-5591] 2022.02.18 11:26:11 [ 11393 ] CreatedReadBufferOrdinary: 6 (increment)
[laptop-5591] 2022.02.18 11:26:11 [ 11393 ] SelectedRows: 1769007 (increment)
[laptop-5591] 2022.02.18 11:26:11 [ 11393 ] SelectedBytes: 24766098 (increment)
[laptop-5591] 2022.02.18 11:26:11 [ 11393 ] ContextLock: 1 (increment)
[laptop-5591] 2022.02.18 11:26:11 [ 11393 ] RealTimeMicroseconds: 257804 (increment)
[laptop-5591] 2022.02.18 11:26:11 [ 11393 ] UserTimeMicroseconds: 210207 (increment)
[laptop-5591] 2022.02.18 11:26:11 [ 11393 ] SystemTimeMicroseconds: 43849 (increment)
[laptop-5591] 2022.02.18 11:26:11 [ 11393 ] SoftPageFaults: 36042 (increment)
[laptop-5591] 2022.02.18 11:26:11 [ 11393 ] OSCPUVirtualTimeMicroseconds: 254050 (increment)
[laptop-5591] 2022.02.18 11:26:11 [ 11393 ] OSReadChars: 12901376 (increment)
[laptop-5591] 2022.02.18 11:26:11 [ 11393 ] OSWriteChars: 11264 (increment)
[laptop-5591] 2022.02.18 11:26:11 [ 11393 ] MemoryTrackerUsage: 69321981 (gauge)
[laptop-5591] 2022.02.18 11:26:11 [ 3017 ] FileOpen: 6 (increment)
[laptop-5591] 2022.02.18 11:26:11 [ 3017 ] ReadBufferFromFileDescriptorRead: 56 (increment)
[laptop-5591] 2022.02.18 11:26:11 [ 3017 ] ReadBufferFromFileDescriptorReadBytes: 7919854 (increment)
[laptop-5591] 2022.02.18 11:26:11 [ 3017 ] ReadCompressedBytes: 8442798 (increment)
[laptop-5591] 2022.02.18 11:26:11 [ 3017 ] CompressedReadBufferBlocks: 345 (increment)
[laptop-5591] 2022.02.18 11:26:11 [ 3017 ] CompressedReadBufferBytes: 20643840 (increment)
[laptop-5591] 2022.02.18 11:26:11 [ 3017 ] OpenedFileCacheMisses: 6 (increment)
[laptop-5591] 2022.02.18 11:26:11 [ 3017 ] IOBufferAllocs: 9 (increment)
[laptop-5591] 2022.02.18 11:26:11 [ 3017 ] IOBufferAllocBytes: 1383231 (increment)
[laptop-5591] 2022.02.18 11:26:11 [ 3017 ] ArenaAllocChunks: 12 (increment)
[laptop-5591] 2022.02.18 11:26:11 [ 3017 ] ArenaAllocBytes: 33546240 (increment)
[laptop-5591] 2022.02.18 11:26:11 [ 3017 ] MarkCacheMisses: 3 (increment)
[laptop-5591] 2022.02.18 11:26:11 [ 3017 ] CreatedReadBufferOrdinary: 6 (increment)
[laptop-5591] 2022.02.18 11:26:11 [ 3017 ] DiskReadElapsedMicroseconds: 4018 (increment)
[laptop-5591] 2022.02.18 11:26:11 [ 3017 ] SelectedRows: 1474188 (increment)
[laptop-5591] 2022.02.18 11:26:11 [ 3017 ] SelectedBytes: 20638632 (increment)
[laptop-5591] 2022.02.18 11:26:11 [ 3017 ] ContextLock: 1 (increment)
[laptop-5591] 2022.02.18 11:26:11 [ 3017 ] RealTimeMicroseconds: 254468 (increment)
[laptop-5591] 2022.02.18 11:26:11 [ 3017 ] UserTimeMicroseconds: 218064 (increment)
[laptop-5591] 2022.02.18 11:26:11 [ 3017 ] SystemTimeMicroseconds: 35821 (increment)
[laptop-5591] 2022.02.18 11:26:11 [ 3017 ] SoftPageFaults: 31287 (increment)
[laptop-5591] 2022.02.18 11:26:11 [ 3017 ] OSCPUWaitMicroseconds: 258 (increment)
[laptop-5591] 2022.02.18 11:26:11 [ 3017 ] OSCPUVirtualTimeMicroseconds: 253879 (increment)
[laptop-5591] 2022.02.18 11:26:11 [ 3017 ] OSReadChars: 7919616 (increment)
[laptop-5591] 2022.02.18 11:26:11 [ 3017 ] OSWriteChars: 10240 (increment)
[laptop-5591] 2022.02.18 11:26:11 [ 3017 ] MemoryTrackerUsage: 63758681 (gauge)
[laptop-5591] 2022.02.18 11:26:11 [ 3018 ] FileOpen: 6 (increment)
[laptop-5591] 2022.02.18 11:26:11 [ 3018 ] ReadBufferFromFileDescriptorRead: 51 (increment)
[laptop-5591] 2022.02.18 11:26:11 [ 3018 ] ReadBufferFromFileDescriptorReadBytes: 11006056 (increment)
[laptop-5591] 2022.02.18 11:26:11 [ 3018 ] ReadCompressedBytes: 10066986 (increment)
[laptop-5591] 2022.02.18 11:26:11 [ 3018 ] CompressedReadBufferBlocks: 396 (increment)
[laptop-5591] 2022.02.18 11:26:11 [ 3018 ] CompressedReadBufferBytes: 24856720 (increment)
[laptop-5591] 2022.02.18 11:26:11 [ 3018 ] OpenedFileCacheMisses: 6 (increment)
[laptop-5591] 2022.02.18 11:26:11 [ 3018 ] IOBufferAllocs: 9 (increment)
[laptop-5591] 2022.02.18 11:26:11 [ 3018 ] IOBufferAllocBytes: 1382713 (increment)
[laptop-5591] 2022.02.18 11:26:11 [ 3018 ] ArenaAllocChunks: 12 (increment)
[laptop-5591] 2022.02.18 11:26:11 [ 3018 ] ArenaAllocBytes: 33546240 (increment)
[laptop-5591] 2022.02.18 11:26:11 [ 3018 ] MarkCacheMisses: 3 (increment)
[laptop-5591] 2022.02.18 11:26:11 [ 3018 ] CreatedReadBufferOrdinary: 6 (increment)
[laptop-5591] 2022.02.18 11:26:11 [ 3018 ] DiskReadElapsedMicroseconds: 4012 (increment)
[laptop-5591] 2022.02.18 11:26:11 [ 3018 ] SelectedRows: 1776007 (increment)
[laptop-5591] 2022.02.18 11:26:11 [ 3018 ] SelectedBytes: 24864098 (increment)
[laptop-5591] 2022.02.18 11:26:11 [ 3018 ] ContextLock: 1 (increment)
[laptop-5591] 2022.02.18 11:26:11 [ 3018 ] RealTimeMicroseconds: 267183 (increment)
[laptop-5591] 2022.02.18 11:26:11 [ 3018 ] UserTimeMicroseconds: 222260 (increment)
[laptop-5591] 2022.02.18 11:26:11 [ 3018 ] SystemTimeMicroseconds: 43835 (increment)
[laptop-5591] 2022.02.18 11:26:11 [ 3018 ] SoftPageFaults: 35873 (increment)
[laptop-5591] 2022.02.18 11:26:11 [ 3018 ] OSCPUWaitMicroseconds: 9 (increment)
[laptop-5591] 2022.02.18 11:26:11 [ 3018 ] OSCPUVirtualTimeMicroseconds: 266084 (increment)
[laptop-5591] 2022.02.18 11:26:11 [ 3018 ] OSReadChars: 10785792 (increment)
[laptop-5591] 2022.02.18 11:26:11 [ 3018 ] OSWriteChars: 11264 (increment)
[laptop-5591] 2022.02.18 11:26:11 [ 3018 ] QueryProfilerRuns: 2 (increment)
[laptop-5591] 2022.02.18 11:26:11 [ 3018 ] MemoryTrackerUsage: 55609403 (gauge)
[laptop-5591] 2022.02.18 11:26:11 [ 3020 ] MemoryTrackerUsage: 0 (gauge)
[laptop-5591] 2022.02.18 11:26:11 [ 3021 ] Seek: 3 (increment)
[laptop-5591] 2022.02.18 11:26:11 [ 3021 ] ReadBufferFromFileDescriptorRead: 61 (increment)
[laptop-5591] 2022.02.18 11:26:11 [ 3021 ] ReadBufferFromFileDescriptorReadBytes: 9781329 (increment)
[laptop-5591] 2022.02.18 11:26:11 [ 3021 ] ReadCompressedBytes: 10299510 (increment)
[laptop-5591] 2022.02.18 11:26:11 [ 3021 ] CompressedReadBufferBlocks: 415 (increment)
[laptop-5591] 2022.02.18 11:26:11 [ 3021 ] CompressedReadBufferBytes: 25231360 (increment)
[laptop-5591] 2022.02.18 11:26:11 [ 3021 ] OpenedFileCacheHits: 3 (increment)
[laptop-5591] 2022.02.18 11:26:11 [ 3021 ] IOBufferAllocs: 9 (increment)
[laptop-5591] 2022.02.18 11:26:11 [ 3021 ] IOBufferAllocBytes: 1386211 (increment)
[laptop-5591] 2022.02.18 11:26:11 [ 3021 ] ArenaAllocChunks: 12 (increment)
[laptop-5591] 2022.02.18 11:26:11 [ 3021 ] ArenaAllocBytes: 33546240 (increment)
[laptop-5591] 2022.02.18 11:26:11 [ 3021 ] MarkCacheHits: 3 (increment)
[laptop-5591] 2022.02.18 11:26:11 [ 3021 ] CreatedReadBufferOrdinary: 5 (increment)
[laptop-5591] 2022.02.18 11:26:11 [ 3021 ] SelectedRows: 1802922 (increment)
[laptop-5591] 2022.02.18 11:26:11 [ 3021 ] SelectedBytes: 25240908 (increment)
[laptop-5591] 2022.02.18 11:26:11 [ 3021 ] ContextLock: 1 (increment)
[laptop-5591] 2022.02.18 11:26:11 [ 3021 ] RealTimeMicroseconds: 266347 (increment)
[laptop-5591] 2022.02.18 11:26:11 [ 3021 ] UserTimeMicroseconds: 218204 (increment)
[laptop-5591] 2022.02.18 11:26:11 [ 3021 ] SystemTimeMicroseconds: 47828 (increment)
[laptop-5591] 2022.02.18 11:26:11 [ 3021 ] SoftPageFaults: 35898 (increment)
[laptop-5591] 2022.02.18 11:26:11 [ 3021 ] OSCPUVirtualTimeMicroseconds: 266022 (increment)
[laptop-5591] 2022.02.18 11:26:11 [ 3021 ] OSWriteBytes: 4096 (increment)
[laptop-5591] 2022.02.18 11:26:11 [ 3021 ] OSReadChars: 9781248 (increment)
[laptop-5591] 2022.02.18 11:26:11 [ 3021 ] OSWriteChars: 10240 (increment)
[laptop-5591] 2022.02.18 11:26:11 [ 3021 ] MemoryTrackerUsage: 64198022 (gauge)
[laptop-5591] 2022.02.18 11:26:11 [ 3019 ] FileOpen: 15 (increment)
[laptop-5591] 2022.02.18 11:26:11 [ 3019 ] Seek: 9 (increment)
[laptop-5591] 2022.02.18 11:26:11 [ 3019 ] ReadBufferFromFileDescriptorRead: 41 (increment)
[laptop-5591] 2022.02.18 11:26:11 [ 3019 ] ReadBufferFromFileDescriptorReadBytes: 8525071 (increment)
[laptop-5591] 2022.02.18 11:26:11 [ 3019 ] ReadCompressedBytes: 8336159 (increment)
[laptop-5591] 2022.02.18 11:26:11 [ 3019 ] CompressedReadBufferBlocks: 318 (increment)
[laptop-5591] 2022.02.18 11:26:11 [ 3019 ] CompressedReadBufferBytes: 20182050 (increment)
[laptop-5591] 2022.02.18 11:26:11 [ 3019 ] OpenedFileCacheMisses: 15 (increment)
[laptop-5591] 2022.02.18 11:26:11 [ 3019 ] IOBufferAllocs: 30 (increment)
[laptop-5591] 2022.02.18 11:26:11 [ 3019 ] IOBufferAllocBytes: 5861226 (increment)
[laptop-5591] 2022.02.18 11:26:11 [ 3019 ] ArenaAllocChunks: 12 (increment)
[laptop-5591] 2022.02.18 11:26:11 [ 3019 ] ArenaAllocBytes: 33546240 (increment)
[laptop-5591] 2022.02.18 11:26:11 [ 3019 ] MarkCacheMisses: 6 (increment)
[laptop-5591] 2022.02.18 11:26:11 [ 3019 ] CreatedReadBufferOrdinary: 18 (increment)
[laptop-5591] 2022.02.18 11:26:11 [ 3019 ] DiskReadElapsedMicroseconds: 42 (increment)
[laptop-5591] 2022.02.18 11:26:11 [ 3019 ] SelectedRows: 1441668 (increment)
[laptop-5591] 2022.02.18 11:26:11 [ 3019 ] SelectedBytes: 20183352 (increment)
[laptop-5591] 2022.02.18 11:26:11 [ 3019 ] ContextLock: 1 (increment)
[laptop-5591] 2022.02.18 11:26:11 [ 3019 ] RealTimeMicroseconds: 245473 (increment)
[laptop-5591] 2022.02.18 11:26:11 [ 3019 ] UserTimeMicroseconds: 194192 (increment)
[laptop-5591] 2022.02.18 11:26:11 [ 3019 ] SystemTimeMicroseconds: 47839 (increment)
[laptop-5591] 2022.02.18 11:26:11 [ 3019 ] SoftPageFaults: 34288 (increment)
[laptop-5591] 2022.02.18 11:26:11 [ 3019 ] OSCPUWaitMicroseconds: 87 (increment)
[laptop-5591] 2022.02.18 11:26:11 [ 3019 ] OSCPUVirtualTimeMicroseconds: 242025 (increment)
[laptop-5591] 2022.02.18 11:26:11 [ 3019 ] OSReadChars: 8122368 (increment)
[laptop-5591] 2022.02.18 11:26:11 [ 3019 ] OSWriteChars: 10240 (increment)
[laptop-5591] 2022.02.18 11:26:11 [ 3019 ] MemoryTrackerUsage: 62068258 (gauge)
[laptop-5591] 2022.02.18 11:26:11 [ 0 ] Query: 1 (increment)
[laptop-5591] 2022.02.18 11:26:11 [ 0 ] SelectQuery: 1 (increment)
[laptop-5591] 2022.02.18 11:26:11 [ 0 ] FileOpen: 32 (increment)
[laptop-5591] 2022.02.18 11:26:11 [ 0 ] Seek: 18 (increment)
[laptop-5591] 2022.02.18 11:26:11 [ 0 ] ReadBufferFromFileDescriptorRead: 269 (increment)
[laptop-5591] 2022.02.18 11:26:11 [ 0 ] ReadBufferFromFileDescriptorReadBytes: 59908385 (increment)
[laptop-5591] 2022.02.18 11:26:11 [ 0 ] ReadCompressedBytes: 57026965 (increment)
[laptop-5591] 2022.02.18 11:26:11 [ 0 ] CompressedReadBufferBlocks: 2223 (increment)
[laptop-5591] 2022.02.18 11:26:11 [ 0 ] CompressedReadBufferBytes: 140000000 (increment)
[laptop-5591] 2022.02.18 11:26:11 [ 0 ] OpenedFileCacheHits: 12 (increment)
[laptop-5591] 2022.02.18 11:26:11 [ 0 ] OpenedFileCacheMisses: 32 (increment)
[laptop-5591] 2022.02.18 11:26:11 [ 0 ] IOBufferAllocs: 75 (increment)
[laptop-5591] 2022.02.18 11:26:11 [ 0 ] IOBufferAllocBytes: 14149190 (increment)
[laptop-5591] 2022.02.18 11:26:11 [ 0 ] ArenaAllocChunks: 79 (increment)
[laptop-5591] 2022.02.18 11:26:11 [ 0 ] ArenaAllocBytes: 201306112 (increment)
[laptop-5591] 2022.02.18 11:26:11 [ 0 ] MarkCacheHits: 18 (increment)
[laptop-5591] 2022.02.18 11:26:11 [ 0 ] MarkCacheMisses: 13 (increment)
[laptop-5591] 2022.02.18 11:26:11 [ 0 ] CreatedReadBufferOrdinary: 44 (increment)
[laptop-5591] 2022.02.18 11:26:11 [ 0 ] DiskReadElapsedMicroseconds: 16091 (increment)
[laptop-5591] 2022.02.18 11:26:11 [ 0 ] NetworkReceiveElapsedMicroseconds: 44 (increment)
[laptop-5591] 2022.02.18 11:26:11 [ 0 ] NetworkSendElapsedMicroseconds: 658 (increment)
[laptop-5591] 2022.02.18 11:26:11 [ 0 ] NetworkSendBytes: 28294 (increment)
[laptop-5591] 2022.02.18 11:26:11 [ 0 ] SelectedParts: 5 (increment)
[laptop-5591] 2022.02.18 11:26:11 [ 0 ] SelectedRanges: 5 (increment)
[laptop-5591] 2022.02.18 11:26:11 [ 0 ] SelectedMarks: 1221 (increment)
[laptop-5591] 2022.02.18 11:26:11 [ 0 ] SelectedRows: 10000000 (increment)
[laptop-5591] 2022.02.18 11:26:11 [ 0 ] SelectedBytes: 140000000 (increment)
[laptop-5591] 2022.02.18 11:26:11 [ 0 ] ContextLock: 88 (increment)
[laptop-5591] 2022.02.18 11:26:11 [ 0 ] RWLockAcquiredReadLocks: 3 (increment)
[laptop-5591] 2022.02.18 11:26:11 [ 0 ] RealTimeMicroseconds: 3756041 (increment)
[laptop-5591] 2022.02.18 11:26:11 [ 0 ] UserTimeMicroseconds: 2856635 (increment)
[laptop-5591] 2022.02.18 11:26:11 [ 0 ] SystemTimeMicroseconds: 295380 (increment)
[laptop-5591] 2022.02.18 11:26:11 [ 0 ] SoftPageFaults: 221523 (increment)
[laptop-5591] 2022.02.18 11:26:11 [ 0 ] OSCPUWaitMicroseconds: 712 (increment)
[laptop-5591] 2022.02.18 11:26:11 [ 0 ] OSCPUVirtualTimeMicroseconds: 3151964 (increment)
[laptop-5591] 2022.02.18 11:26:11 [ 0 ] OSWriteBytes: 4096 (increment)
[laptop-5591] 2022.02.18 11:26:11 [ 0 ] OSReadChars: 59908096 (increment)
[laptop-5591] 2022.02.18 11:26:11 [ 0 ] OSWriteChars: 65536 (increment)
[laptop-5591] 2022.02.18 11:26:11 [ 0 ] QueryProfilerRuns: 9 (increment)
[laptop-5591] 2022.02.18 11:26:11 [ 0 ] MemoryTrackerUsage: 219393274 (gauge)
[laptop-5591] 2022.02.18 11:26:11 [ 3021 ] DiskReadElapsedMicroseconds: 4019 (increment)
[laptop-5591] 2022.02.18 11:26:11 [ 3022 ] DiskReadElapsedMicroseconds: 4000 (increment)
[laptop-5591] 2022.02.18 11:26:11 [ 3019 ] MarkCacheHits: 6 (increment)
[laptop-5591] 2022.02.18 11:26:11 [ 3022 ] OSCPUWaitMicroseconds: 16 (increment)
[laptop-5591] 2022.02.18 11:26:11 [ 3017 ] QueryProfilerRuns: 2 (increment)
[laptop-5591] 2022.02.18 11:26:11 [ 3021 ] FileOpen: 2 (increment)
[laptop-5591] 2022.02.18 11:26:11 [ 11393 ] FileOpen: 3 (increment)
[laptop-5591] 2022.02.18 11:26:11 [ 3021 ] MarkCacheMisses: 1 (increment)
[laptop-5591] 2022.02.18 11:26:11 [ 3019 ] OpenedFileCacheHits: 3 (increment)
[laptop-5591] 2022.02.18 11:26:11 [ 3021 ] OpenedFileCacheMisses: 2 (increment)
[laptop-5591] 2022.02.18 11:26:11 [ 11393 ] OpenedFileCacheMisses: 3 (increment)
[laptop-5591] 2022.02.18 11:26:11 [ 3019 ] QueryProfilerRuns: 1 (increment)
[laptop-5591] 2022.02.18 11:26:11 [ 11393 ] QueryProfilerRuns: 1 (increment)
```

**Comment by alexey-milovidov** [comment]
@azat the value `-1` looks idiosynchratic to me.

**Comment by alexey-milovidov** [comment]
And the whole feature does not look like it should be used at all. It's just a something private from @azat

**Comment by azat** [comment]
>@azat the value -1 looks idiosynchratic to me.

Actually I would say that it is the only value that makes sense, since otherwise there is too much information (of course if you don't parse those profile events and interpret it in a more human readable way, like vmstat or similar)

>And the whole feature does not look like it should be used at all. It's just a something private from @azat

Yes, this is mostly for debugging, but can be useful, since this is the way to get those events w/o entry in query_log.

Anyway this particular problem should be addressed by #34749
